core/privacy/test/zokrates.spec.ts

```diff
@@ -0,0 +1,21 @@
+import { shouldBehaveLikeZKSnarkCircuit } from './shared';
+import { zkSnarkCircuitProviderServiceFactory, zkSnarkCircuitProviderServiceZokrates } from '../src/index';
+
+const baselineDocumentCircuitPath = '../../lib/circuits/baselineDocument.zok';
+const noopAgreementCircuitPath = '../../lib/circuits/noopAgreement.zok';
+
+let provider;
+
+describe('when the underlying zokrates provider is available', () => {
+  beforeEach(async () => {
+    provider = await zkSnarkCircuitProviderServiceFactory(zkSnarkCircuitProviderServiceZokrates);
+  });
+
+  it('runs the noopAgreement circuit lifecycle successfully', () => {
+    shouldBehaveLikeZKSnarkCircuit(provider, noopAgreementCircuitPath); // this is here for sanity...
+  });
+
+  it('runs the baselineDocument circuit lifecycle successfully', () => {
+    shouldBehaveLikeZKSnarkCircuit(provider, baselineDocumentCircuitPath); // this takes... a while... ;)
```

Comment:
This test suite hangs (for >15min) at the trusted setup step:
```
  circuit: ../../lib/circuits/baselineDocument.zok
    compile
      ✓ should have output the compiled circuit
      ✓ should have output the ABI of the compiled circuit
      trusted setup
```

Reply:
The setup for this circuit takes quite a while. Going to use a different one while `baselineDocument` continues to be optimized.

Reply:
updating an alternate method to fetch pre-set up circuits and associated artifacts